applications/luci-app-statistics/luasrc/statistics/datatree.lua

```diff
@@ -17,13 +16,9 @@ function Instance.__init__( self, host )
 	self._libdir  = "/usr/lib/collectd"
 	self._rrddir  = "/tmp/rrd"
 
-	if sections and sections.collectd then
-		self._host    = host or sections.collectd.Hostname or sys.hostname()
-		self._libdir  = sections.collectd.PluginDir        or "/usr/lib/collectd"
-	end
-	if sections and sections.collectd_rrdtool then
-		self._rrddir  = sections.collectd_rrdtool.DataDir  or "/tmp/rrd"
-	end
+	self._host    = host or uci:get("collectd", "globals", "hostname") or sys.hostname()
+	self._libdir  = uci:get("collectd", "globals", "PluginDir") or "/usr/lib/collectd"
+	self._rrddir  = uci:get("collectd", "rrdtool", "DataDir") or "/tmp/rrd"
```

Comment:
Please elaborate why do you see the safeguards added by d04f667383 can be now removed.

Reply:
Thanks for your feedback. I think this is not neccessary because if the section is not present it will return **nil**. But this is only assumption I will have to check this. Thanks for the hint.

Reply:
So I have checked it.

1. If collectd config file is not present then the default value after **or** is used because the uci:get returned nil
2. If the rrdtool config section is not presented then the default value after **or** is used because the uci:get returned nil too
3. If the rrdtool section does not have a the option then the default value after **or** is used as well because uci:get returned nil.

So I think this is not necessary anymore because we use a function call from **uci:get** and do not use a variable which could be nil.